fastify.js

```diff
@@ -773,12 +773,23 @@ function build (options) {
   function setNotFoundHandler (opts, handler) {
     throwIfAlreadyStarted('Cannot call "setNotFoundHandler" when fastify instance is already started!')
 
+    const _fastify = this
     const prefix = this._routePrefix || '/'
 
     if (this._canSetNotFoundHandler === false) {
       throw new Error(`Not found handler already set for Fastify instance with prefix: '${prefix}'`)
     }
 
+    if (typeof opts === 'object' && opts.beforeHandler) {
+      if (Array.isArray(opts.beforeHandler)) {
```

Comment:
Maybe I'm missing something here, why this should be an array?

Reply:
You can pass a function (only one beforeHandler) or an array of functions (with multiple beforeHandlers).

Reply:
My bad, well done!